File: 2020-11-16 stop-gninnips-my-sdrow/solution.js
// solution to "Stop gninnipS My sdroW!"

function spinWords(sentence){
    // "words" is the sentence split into words
    // "output" is the output string
    let words = sentence.split(" ");
    let output = ""
    
    // function to reverse the string
    // splits the string to an array of characters, 
    // reverses their order, and joins them back
    // returns the reversed string
    function reverse(str) {
        let split_str = str.split("");
        let reverse_str = split_str.reverse();
    
        return reverse_str.join("");
    }
    
    // appends reversed word if the length is greater than 5 
    // otherwise. append unchanged word
    words.forEach(word => {
        (word.length >= 5) ? (output += (reverse(word) + " ")) : (output += (word + " "))
    });

    // removes trailing space
    return output.slice(0, -1);
}

File: 2020-11-16 stop-gninnips-my-sdrow/solution.py
# solution to "Stop gninnipS My sdroW!"
def spin_words(sentence):
    # "spin": input is split into a list of words
    # "output": output stream
    spin = sentence.split(" ")
    output = ""

    # add reversed word if length >= 5
    # otherwise add normal word
    for word in spin:
        if len(word) >= 5:
            output += word[::-1]
        else:
            output += word

        output += " "
    
    # return string with space ommited
    return output[:-1]